File: backend/src/models/User.model.ts
import mongoose, { Schema, Document, Types, model } from "mongoose";
import { IUser } from '../interfaces';

export type IUserDocument = IUser & Document;

export const UserSchema = new Schema({
    pseudo: {
        type: String,
        required: true,
        unique: true
    },
    idFirebase: {
        type: String,
        required: true
    },
    idImage: {
        type: String,
        required: true
    },
    friends: [
        {
            type: Types.ObjectId,
            required: true,
            ref: 'User',
            default: []
        }
    ],
    places: [
        {
            type: Types.ObjectId,
            required: true,
            ref: 'Place',
            default: []
        }
    ],
    lastConnexion: {
        type: Date,
        required: true,
        default: new Date((new Date()).getFullYear(), (new Date()).getMonth(), (new Date()).getDate() - 1),  
    },
},
{
    versionKey: false,
    timestamps: true
});

export const UserModel = model<IUserDocument>('User', UserSchema);

File: frontend/la_map/lib/models/user_model.dart
class ApiUser {
  final String pseudo;
  final String id;
  final String idFirebase;
  final String idImage;
  final List<dynamic> friends;
  //final DateTime lastConnexion;

  const ApiUser({
    required this.pseudo,
    required this.id,
    required this.idFirebase,
    required this.friends,
    required this.idImage,
  });

  factory ApiUser.fromJson(Map<String, dynamic> json) {
    return ApiUser(
      pseudo: json['pseudo'],
      id: json['_id'],
      friends: json['friends'],
      idImage: json['idImage'],
      idFirebase: json['idFirebase'],
    );
  }
}
